packages/app/components/claim/claim-form.tsx

```diff
@@ -71,6 +78,25 @@ export const ClaimForm = ({ edition }: { edition: CreatorEditionResponse }) => {
   //     });
   // }, [web3]);
 
+  if (!isAuthenticated) {
+    return (
+      <View tw="p-4">
+        <Button onPress={navigateToLogin}>Please login to continue</Button>
+      </View>
+    );
+  }
+
+  // TODO: remove this after imperative login modal API in rainbowkit
+  if (!connected) {
```

Comment:
I guess we are gonna need to update connected conditions because if the user login with email and refresh that page, they will be in this status: authenticated but not connected (should be connected). I'll do some checks and push to this branch

Reply:
yes, good catch. actually, I didn't handle that because it was a bit unsure whether we're supporting magic wallet for claiming. Feel free to add it!